### PR TITLE
[FLINK-11367][tests] Port TaskManagerProcessReapingTestBase to new codebase

### PR DESCRIPTION
**Pull request by GJL**
## What is the purpose of the change

*This ports the tests around `TaskManagerProcessReapingTestBase` to  the FLIP-6 codebase.*

## Brief change log

  - *Delete `TaskManagerProcessReapingFatalErrorTest`*
  - *Delete `TaskManagerProcessReapingTest`*
  - *Delete `TaskManagerProcessReapingTestBase`*
  - *Add `TaskManagerRunner#testShouldShutdownOnFatalError`*

## Verifying this change

This change added tests and can be verified as follows:

  - *Added test `TaskManagerRunner#testShouldShutdownOnFatalError`*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerTest.java [hunk 1]
```diff
@@ -37,44 +43,57 @@
...
+		try (TaskManagerRunner taskManagerRunner = createTaskManagerRunner(configuration)) {
+			taskManagerRunner.start();
 
-		public TestingTaskManagerRunner(Configuration configuration, ResourceID resourceId, CompletableFuture<Void> jvmTerminationFuture) throws Exception {
-			super(configuration, resourceId);
-			this.jvmTerminationFuture = jvmTerminationFuture;
-		}
+			final SystemExitTrackingSecurityManager systemExitTrackingSecurityManager =
+				runWithSystemExitTracking(() ->
+					taskManagerRunner.getTerminationFuture().get());
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerTest.java]: shouldn't we call `taskManagerRunner.start();` inside this runnable as well? If the runnable exits before `runWithSystemExitTracking` is called, will it still track `exit` calls?

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerTest.java [hunk 2]
```diff
@@ -37,44 +43,57 @@
...
-		configuration.setString(JobManagerOptions.ADDRESS, localhost.getInetAddress().getHostName());
-		configuration.setInteger(JobManagerOptions.PORT, localhost.getLocalPort());
-		configuration.setString(TaskManagerOptions.REGISTRATION_TIMEOUT, "10 ms");
-		final CompletableFuture<Void> jvmTerminationFuture = new CompletableFuture<>();
-		final TestingTaskManagerRunner taskManagerRunner = new TestingTaskManagerRunner(configuration, taskManagerResourceId, jvmTerminationFuture);
+	@Rule
+	public final Timeout timeout = Timeout.seconds(30);
 
-		taskManagerRunner.start();
+	private Configuration configuration;
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerTest.java]: having this as a field seems unnecessary; I'd rather have a method that creates and returns this configuration.

**Comment by zentol** [comment]
@flinkbot approve descriptions
@flinkbot approve consensus

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerTest.java (hunk 1 above)
Comment (GJL) [reply flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerTest.java]: Yes, should be inside.

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerTest.java (hunk 2 above)
Comment (GJL) [reply flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerTest.java]: Ok, changed.

**Comment by GJL** [comment]
Merging...

**Pull request closed without merging**